### ISSUE TITLE
General question - are attachments rendered? JPGs etc.

### ISSUE THREAD

**Issue by james-cook** [newroco/emlviewer#12]
The EML files do appear as shown - but  attachments such as JPGs are not rendered.
Is this the expected behaviour?

(I "assume" that the EML format embeds attachments in some way)
(Tried NC15 / apache and NC18 / nginx)

**Comment by putt1ck** [comment]
The assumption is that the attachment would have already been dealt with in a project context; this is just a way to view emails that have been stored alongside other project documents (emails being an element often missing from many project file stores, maybe because they were hard to view having been stored there).

I imagine making a viewer that made attachments clickable would be possible, but I think I would look first at the ingest method; so assuming this is an automated scenario where emails are being delivered directly to the NC instance, I would think it would be preferable to strip and store the email body as one item and the attachment as another. I would then want the email to show it had an attachment name of X, so a little post-processing required; if this was an auditable process I'd want to show a trail for that, up to also storing the email in original received form - but would still want to separate them for ease of access within the file store.

**Comment by lucianpricop** [comment]
I would like to add that images referenced in eml files stored on external servers, are blocked by nextcloud policy by default. An admin can allow this, changing certain configuration parameters, but this is not advisable.

**Comment by putt1ck** [comment]
Not only not advisable but in the use case of emails stored for a project archive, the chance of an external URL staying constant over time is pretty low. It's the email (text) content that's most important to be able to share/view in the context of other project files that's the neat bit.

@james-cook do you have a different use case scenario?

**Comment by james-cook** [comment]
My particular use case is archiving the life of a departed relative. Gmail/imap exported to eml format from thunderbird.
The eml file names are configured to show date, subject, sender, recipient.

In the first case I just want to be able to store and display eml files (later, showing conversations would be great(!)).

I am not an expert on the eml format.
In my case the attachments/jpegs do seem to be embedded in base 64 - but these images don't appear in the preview (in my setup at least)

**Comment by putt1ck** [comment]
We could definitely look at a way to parse the attachments in the preview - I guess they would have to be links which can be used to download the attachment though.

**Comment by lucianpricop** [comment]
> My particular use case is archiving the life of a departed relative. Gmail/imap exported to eml format from thunderbird.
> The eml file names are configured to show date, subject, sender, recipient.
> 
> In the first case I just want to be able to store and display eml files (later, showing conversations would be great(!)).
> 
> I am not an expert on the eml format.
> In my case the attachments/jpegs do seem to be embedded in base 64 - but these images don't appear in the preview (in my setup at least)

Hello, we just made a new release and it now allows external or embedded images. Can you please test if it works fine in your case? use version 0.0.10 Thanks

**Comment by james-cook** [comment]
I loaded emlviewer 12 and tested (I upgraded my NC15 to NC17)
Side note: I enabled the tidy mod (mbstring was already present)

The embedded images do not render...

Just to be sure here is a "sample" email with private stuff overwritten. This is the sort of eml I am hoping to render:
`
MIME-Version: 1.0
X-Received: by 10.112.29.36 with SMTP id g4mr3111152lbh.56.1428082427952; Fri,
 03 Apr 2015 10:33:47 -0700 (PDT)
Received: by 10.112.78.72 with HTTP; Fri, 3 Apr 2015 10:33:46 -0700 (PDT)
Received: by 10.112.78.72 with HTTP; Fri, 3 Apr 2015 10:33:46 -0700 (PDT)
Date: Fri, 3 Apr 2015 18:33:46 +0100
.
.
.
Message-ID: <XXX@mail.gmail.com>
Subject: 
From: XXX
To: XXX
Content-Type: multipart/mixed; boundary=001a1133f994d405090512d559bc

--001a1133f994d405090512d559bc
Content-Type: multipart/alternative; boundary=001a1133f994d405040512d559ba

--001a1133f994d405040512d559ba
Content-Type: text/plain; charset=UTF-8

Hi XXX,

Love
YYY

--001a1133f994d405040512d559ba
Content-Type: text/html; charset=UTF-8
<p dir="ltr">Hi XXX</p>
YYY</p>

--001a1133f994d405040512d559ba--
--001a1133f994d405090512d559bc
Content-Type: image/jpeg; name="20150403_183233.jpg"
Content-Disposition: attachment; filename="20150403_183233.jpg"
Content-Transfer-Encoding: base64
X-Attachment-Id: 1497452578085011456-local0

/9j/4VkRRXhpZgAASUkqAAgAAAAMAAABBAABAAAAIBAAAAEBBAABAAAAEgkAAA8BAgAIAAAAngAA
ABABAgAJAAAApgAAABIBAwABAAAABgAAABoBBQABAAAAsAAAABsBBQABAAAAuAAAACgBAwABAAAA
AgAAADEBAgAOAAAAwAAAADIBAgAUAAAAzgAAABMCAwABAAAAAQAAAGmHBAABAAAA4gAAAG4TAABT
QU1TVU5HAEdULUk5NTE1AABIAAAAAQAAAEgAAAABAAAASTk1MTVYWFUxQU5EMwAyMDE1OjA0OjAz
IDE4OjMyOjMzAB8AmoIFAAEAAABcAgAAnYIFAAEAAABkAgAAIogDAAEAAAACAAAAJ4gDAAEAAAAy
.
.
.
4Vem7/UVRX76/U/yNBlPcvqACMAdfSp/+Wkf0/qahXqPrU3/AC0j+n9TQOmTwjLjPOWAOa1bYDfJ
x/Hg/lWVD98f7wrVtvvv/v8A9KDQ/9f+FUD97IPTOPyrqrAB7K53gPjaRuGcc1yq/wCuk/H+VdXp
3/Hlc/Rf516AHHahxdTY4+VRx9TWG5JaQEkgPwCenFbmo/8AH1N/ur/M1hP9+T/fH8qAGY/Sr6E/
Z15PcdfrVCryf8e6/j/WgCuOQmf7pH60knzAbueR1pV6J/un+dI/QfWgD//Z
--001a1133f994d405090512d559bc--


`

**Comment by putt1ck** [comment]
I think the new functionality specifically covers embedded images, where someone has them inline within HTML emails. It wouldn't try to decode attachments - there's a bunch of workflow challenges associated with that idea (where do you put the attachment once extracted (data duplication)? Do you delete or keep after the viewer closes? How do you present in the viewer - given it can't deal with all attachment types?).

For images you want to be able to view it might be a better path to strip them out with an EML attachment extractor and them load them into a folder in your NC, which would allow you to view with gallery (and then have the option to share the folder with others who might be interested).

**Comment by james-cook** [comment]
I'm not a NC platform expert - isn't this "just" a case of rendering an existing file (the *.eml file) in a particular way? No need to store it separately? Perhaps just only the first image if many are a problem...

FYI I have extracted the images too - but it would be nice to have them offered in the context of the email in some way.
